client/my-sites/domains/domain-management/settings/cards/registered-domain-details.tsx

```diff
@@ -122,7 +122,7 @@ const RegisteredDomainDetails = ( {
 			<RenewButton
 				purchase={ purchase }
 				selectedSite={ selectedSite }
-				subscriptionId={ parseInt( domain.subscriptionId!, 10 ) }
+				subscriptionId={ parseInt( domain.subscriptionId ?? '', 10 ) }
```

Comment:
no-null assertions are banned by our linting rules, and they are dangerous anyway. By adding a fallback we provide the same silent failure without needing to override TypeScript.